app/javascript/components/ansible-credentials-form/ansible-credential-form.schema.js

```diff
@@ -0,0 +1,39 @@
+import { componentTypes } from '@@ddf';
+import { ForkJoinObservable } from 'rxjs/observable/ForkJoinObservable';
+import { API } from '../../http_api';
+
+const createSchema = (fields, setFields) => ({
+  fields: [
+    {
+      component: componentTypes.TEXT_FIELD,
+      id: 'name',
+      name: 'name',
+      label: __('Name'),
+      maxLength: 128,
+    },
+    {
+      component: 'select-with-onchange',
+      id: 'credentialType',
+      name: 'credentialType',
+      label: __('Credential type'),
+      placeholder: __('<Choose>'),
+      onChange: value => API.options('/api/authentications').then(({ data: { credential_types: { embedded_ansible_credential_types } } }) =>
```

Comment:
As you're calling the same `API.options` twice, this can be cached, but instead of storing it in the variable, you could use the implicit caching of the promise. If you extract the promise into a variable in the `AnsibleCredentialsForm` and pass it as an argument to `createSchema`, you can safely call `then` on the promise twice.

This implicit caching would not survive a react rerender, but that can be fixed by wrapping it with a [`useMemo`](https://reactjs.org/docs/hooks-reference.html#usememo) hook. Something like:
```js
import React, { useState, useMemo } from 'react';

// ...

const AnsibleCredentialsForm = () => { 
  // ..
  const promise = useMemo(() => API.options('/api/authentications'), []);

  return (
    <MiqFormRenderer
      schema={createSchema(fields, setFields, promise)}
      // ...
  );
};
```

And then in the `createSchema` function here:
```js
const createSchema = (fields, setFields, promise) => ({
  fields: [
    // ...
    {
      component: 'select-with-onchange',
      // ...
      onChange: value => promise.then( // ...
      loadOptions: () => promise.then( // ...
    }
  ]
});
```